source/shared/autofill.js

```diff
@@ -1,74 +1,104 @@
 import { NativeModules, Platform } from "react-native";
 import { EntryFinder } from "../library/buttercupCore";
+import { getEntryPathString } from "./entry";
 
 const { AutoFillBridge } = NativeModules;
 
-export function addSourceToAutoFill(sourceID, archive) {
+// Note: AutoFillBridge is not available in jest, and cannot be mocked without mocking
+//  every single NativeModule from node_modules as well (not something I have time to do)
+//  Instead, just check for both the existence of AutoFillBridge and the value of DEVICE_SUPPORTS_AUTOFILL
+//  - @se1exin 17/2/19
+export const autoFillAvailable = AutoFillBridge && !!AutoFillBridge.DEVICE_SUPPORTS_AUTOFILL;
+
+export function getAutoFillSystemStatus() {
+    return new Promise((resolve, reject) => {
+        if (autoFillAvailable) {
+            return AutoFillBridge.getAutoFillSystemStatus().then(isAutoFillEnabled => {
```

Comment:
Same here

Reply:
Added missing resolve()

Reply:
I mean it's alright of course, it's just a bit weird seeing the promise being returned when using the `new Promise((resolve, reject) => ...` format. I'm fine with this as it works of course.

Reply:
Alternatively (maybe):

```
export function getAutoFillSystemStatus() {
  return autoFillAvailable ? AutoFillBridge.getAutoFillSystemStatus() : Promise.resolve(false);
}
```